frontend/src/layouts/default.vue

```diff
@@ -11,16 +11,26 @@
 
 <script setup lang="ts">
 import { ref, watch, provide } from 'vue';
+import { useRoute } from 'vue-router';
 import { useDisplay } from 'vuetify';
 import { userLibrariesStore } from '@/store';
 
 const display = useDisplay();
+const route = useRoute();
 const userLibraries = userLibrariesStore();
 const navDrawer = ref(!display.mobile.value);
 
-if (!userLibraries.isReady) {
-  await userLibraries.refresh();
-}
+await userLibraries.refresh();
+
+watch(
+  () => route.name,
+  (newValue) => {
+    // @ts-expect-error - index isn't part of route values
+    if (newValue === 'index') {
+      userLibraries.refresh();
+    }
```

Comment:
```suggestion
  async (newValue) => {
    // @ts-expect-error - index isn't part of route values
    if (newValue === 'index') {
      await userLibraries.refresh();
    }
```

I'd also add a comment like this:

```
Loading recent items from libraries can take a long time from server. To improve UX, we block navigation to the default layout when loading the client, so all the content is present when the page is finally rendered, avoiding content jumping.

We need to use a top-level await to block navigation, but we also want to ensure content is up-to-date when accessing the page (and that's why we trigger a refetch when accessing index as well).
```